content/en/blog/2023/egress-sni/index.md

```diff
@@ -0,0 +1,441 @@
+                stat_prefix: tcp
+                cluster: dynamic_forward_proxy_cluster
+                access_log:
+                - name: envoy.access_loggers.file
+                  typed_config:
+                    "@type": type.googleapis.com/envoy.extensions.access_loggers.file.v3.FileAccessLog
+                    path: "/dev/stdout"
+                    log_format:
+                      text_format_source:
+                        inline_string: '[%START_TIME%] "%REQ(:METHOD)% %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)%
+                          %PROTOCOL%" %RESPONSE_CODE% %RESPONSE_FLAGS% %RESPONSE_CODE_DETAILS% %CONNECTION_TERMINATION_DETAILS%
+                          "%UPSTREAM_TRANSPORT_FAILURE_REASON%" %BYTES_RECEIVED% %BYTES_SENT% %DURATION%
+                          %RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)% "%REQ(X-FORWARDED-FOR)%" "%REQ(USER-AGENT)%"
+                          "%REQ(X-REQUEST-ID)%" "%REQ(:AUTHORITY)%" "%UPSTREAM_HOST%" %UPSTREAM_CLUSTER%
+                          %UPSTREAM_LOCAL_ADDRESS% %DOWNSTREAM_LOCAL_ADDRESS% %DOWNSTREAM_REMOTE_ADDRESS%
+                          %REQUESTED_SERVER_NAME% %ROUTE_NAME%
+
+                          '
+  - applyTo: NETWORK_FILTER
+    match:
+      context: GATEWAY
+      listener:
+        filterChain:
+          filter:
+            name: "envoy.filters.network.tcp_proxy"
+    patch:
+      operation: MERGE
+      value:
+        name: envoy.tcp_proxy
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.network.tcp_proxy.v3.TcpProxy
+          stat_prefix: tcp
+          cluster: sni_cluster
+{{< /text >}}
+
+Check the logs of `istiod` and the gateway pod for any errors of warnings. If all goes well, your mesh sidecars are now routing
+`*.wikipedia.org` towards your gateway pod, while the gateway pod is routing to the exact remote host based on the application
+request.
+
+## Testing
+
+Following other examples, we will use the sleep pod. Assuming there is automatic sidecar injection in your default namespace, deploy
+the test app:
+
+{{< text bash >}}
+$ kubectl apply -f {{< github_file >}}/samples/sleep/sleep.yaml
+{{< /text >}}
+
+Get your pod and gateway name:
+
+{{< text bash >}}
+$ export SOURCE_POD=$(kubectl get pod -l app=sleep -o jsonpath={.items..metadata.name})
+$ export GATEWAY_POD=$(kubectl get pod -n istio-egress -l istio=egressgateway -o jsonpath={.items..metadata.name})
+{{< /text >}}
+
+Test that you are able to connect to a `wikipedia.org` site:
+
+{{< text bash >}}
+$ kubectl exec "$SOURCE_POD" -c sleep -- sh -c 'curl -s https://en.wikipedia.org/wiki/Main_Page | grep -o "<title>.*</title>"; curl -s https://de.wikipedia.org/wiki/Wikipedia:Hauptseite | grep -o "<title>.*</title>"'
+<title>Wikipedia, the free encyclopedia</title>
+<title>Wikipedia – Die freie Enzyklopädie</title>
+{{< /text >}}
+
+We could reach both English and German Wikipedia, great! To have something to compare, let's check another site:
```

Comment:
```suggestion
We could reach both English and German `wikipedia.org` subdomains, great!

Normally, in a production environment, we would [block external requests](/docs/tasks/traffic-management/egress/egress-control/#change-to-the-blocking-by-default-policy) that are not configured to redirect through the egress gateway, but since we didn't do that in our test environment, let's access another external site for comparison:
```